File: go/arrays/contains_duplicate_test.go
package arrays

import "testing"

func TestContainsDuplicate(t *testing.T) {
	tests := []struct {
		name     string
		nums     []int
		expected bool
	}{
		{"has duplicate", []int{1, 2, 3, 1}, true},
		{"no duplicate", []int{1, 2, 3, 4}, false},
		{"all same", []int{1, 1, 1, 1}, true},
		{"empty", []int{}, false},
		{"single element", []int{1}, false},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := ContainsDuplicate(tt.nums)
			if result != tt.expected {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}

File: typescript/arrays/containsDuplicate.test.ts
import { containsDuplicate } from './containsDuplicate';

describe('containsDuplicate', () => {
  it('should return true if array has duplicates', () => {
    expect(containsDuplicate([1, 2, 3, 1])).toBe(true);
  });

  it('should return false if array has no duplicates', () => {
    expect(containsDuplicate([1, 2, 3, 4])).toBe(false);
  });

  it('should return true if all elements are same', () => {
    expect(containsDuplicate([1, 1, 1, 1])).toBe(true);
  });

  it('should return false for empty array', () => {
    expect(containsDuplicate([])).toBe(false);
  });

  it('should return false for single element', () => {
    expect(containsDuplicate([1])).toBe(false);
  });
});
